src/primitives.coffee

```diff
@@ -27,6 +27,61 @@ class ElementType
 
   set: (@content) -> @
 
+  getProperty: -> new ErrorType 'Element does not have method getProperty', @
+
+  has: -> false
+
+  map: -> new ErrorType 'Element does not have method map', @
+
+  filter: -> new ErrorType 'Element does not have method filter', @
+
+  forEach: -> new ErrorType 'Element does not have method forEach', @
+
+  length: -> new ErrorType 'Element does not have method length', @
+
+  push: -> new ErrorType 'Element does not have method push', @
+
+  add: -> new ErrorType 'Element does not have method add', @
+
+  find: -> new ErrorType 'Element does not have method find', @
+
+  keys: -> new ErrorType 'Element does not have method keys', @
+
+  values: -> new ErrorType 'Element does not have method values', @
+
+class ErrorType
```

Comment:
Shouldn't toValue and other things be implement for polymorphism? Maybe they should raise an error or even something like raise @ and have ErrorType inherit from Error (that might be interesting). Then basically, you might solve my concern that when you try to use the ErrorType, it throws itself and gives a useful traced message or something.